toolboxes/mri/pmri/gpu/cuNonCartesianSenseOperator.cu

```diff
@@ -48,11 +50,47 @@ cuNonCartesianSenseOperator<REAL,D>::mult_MH( cuNDArray< complext<REAL> >* in, c
 	  throw std::runtime_error("cuNonCartesianSenseOperator::mult_MH: input/output arrays do not match specified domain/codomains");
   }
   std::vector<size_t> tmp_dimensions = *this->get_domain_dimensions();
+  std::vector<size_t> tmp_dimensions_data = *this->get_codomain_dimensions();
+
+  
+  auto RO = in->get_size(0);
+  auto E1E2 = in->get_size(1);
+  auto CHA = in->get_size(2);
+  
   tmp_dimensions.push_back(this->ncoils_);
   cuNDArray< complext<REAL> > tmp(&tmp_dimensions);
 
- // Do the NFFT
-  plan_->compute( *in, tmp, dcw_.get(), NFFT_comp_mode::BACKWARDS_NC2C );
+  // Remove channel dimension if the last dimension is the same as the number of coils
+   if(tmp_dimensions_data[tmp_dimensions_data.size()-1]==this->ncoils_ && tmp_dimensions_data.size()>2)
+   {
+     tmp_dimensions_data.pop_back(); 
+     cuNDArray<complext<REAL>> temp_ch_data(&tmp_dimensions_data);
+     tmp_dimensions.pop_back();
+     cuNDArray<complext<REAL>> temp_ch_recon(&tmp_dimensions);
+
+
+      for (int iCHA = 0; iCHA < CHA; iCHA++)
+        {
```

Comment:
This needs a bit more explanation. And you need to verify that it is also faster for 2D, and not just 3D